ui/lib/sync/addon/utils/api-transforms.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { findDestination } from 'core/helpers/sync-destinations';
+
+import type { DestinationType, ListDestination } from 'vault/sync';
+import type { SystemListSyncDestinationsResponse } from '@hashicorp/vault-client-typescript';
+
+export const listDestinationsTransform = (
+  response: SystemListSyncDestinationsResponse,
+  nameFilter?: string,
+  typeFilter?: string
+) => {
+  const { keyInfo } = response;
+  const destinations: ListDestination[] = [];
+  // loop through each destination type (keys in key_info)
+  for (const key in keyInfo) {
+    // iterate through each type's destination names
+    const names = (keyInfo as Record<string, string[]>)[key];
+    names?.forEach((name: string) => {
+      // remove trailing slash from key
+      const type = key.replace(/\/$/, '') as DestinationType;
+      const id = `${type}/${name}`;
+      const { icon, name: typeDisplayName } = findDestination(type) || {};
+      // create object with destination's id and attributes
+      destinations.push({ id, name, type, icon, typeDisplayName });
+    });
```

Comment:
```suggestion
    const names = keyInfo[rawType] ?? [];
    const type = key.replace(/\/$/, '') as DestinationType;
    
    names.forEach((name: string) => {
      const id = `${type}/${name}`;
      const { icon, name: typeDisplayName } = findDestination(type) || {};
      // create object with destination's id and attributes
      destinations.push({ id, name, type, icon, typeDisplayName });
    });
```

Reply:
This suggestion fails type checking. I can move the type const assignment out of the loop though.